### DISCUSSION TITLE
Bungee wide player balance

### DISCUSSION THREAD

**Discussion by WarWolf-1** [Zrips/CMI#5540] [Enhancement]
It would be really nice to have support for synced balance across bungee. Can it be done?

**Comment by CloudeLecaw** [comment]
U might should use the mysql database for cmi Player file that all Our servers of your bungeen have access too. This way all servers of you bungee have access to one balance (and other information)

What will not work is using the default sqlite database. Your Server will have no access to it (only the one where the file is stored)

To enable MySQL simply enable it in the database.yml file in the cmi folder.

**Comment by WarWolf-1** [comment]
Database config file clearly advises against sharing databases
"# ATTENTION! DON'T USE SAME DATABASE TABLES FOR MORE THEN ONE SERVER, YOU WILL HAVE UNEXPECTED ISSUES. EACH SERVER SHOULD HAVE ITS OWN DATABASE TABLE SETUP"

**Comment by CloudeLecaw** [comment]
Sharing means not using them at the same time. It Different if your one player that has one Connection to a database.
Its a good solution. Without mysql u should not think of bungee anymore ;)

What is meant here that not two servers should use one database at the same time.
Example:
You have a mysql and give me your data.
I setup anserver that uses These mysql.
Good: we share all like balance
Bad: if Player do Something on my server then they *do" the same onnyours.
But in your case the bungee Network is count as one whole server

**Comment by WarWolf-1** [comment]
So you're saying it is ok to share the cmi database between bungee servers because the player is only connected to one of the servers?

**Comment by WarWolf-1** [comment]
But if 2 servers can't use the database at the same time, then the player balance can not be synced between 2 servers...

**Comment by CloudeLecaw** [comment]
U need an example now why its not possible?:)
- First the player cant be logged in at two servers at once.
- when he joins server a and sets 29  homes and give hisself 20000 milliin and leave
Then he also has 29 homes and 20000 millin on your server that might hasba total other concept
If you count one bunggee Network where the network itself is a whole server. Therr is no reason for notnsharing. The player still will be only one server. Abd if he has 20000 milln on serverpart a it might be wanted he also has it on serverpart b

What we learned:  share your database on your bungee, but DONT shate it with servers that are no part of the bungee.

**Comment by WarWolf-1** [comment]
But what if I only want to share balance and not other data like homes?

**Comment by CloudeLecaw** [comment]
This Share all and just use balance on other part of your bungee
Its a binary thing.. all or nothing...1 or 0

**Comment by WarWolf-1** [comment]
Understandable, but maybe dev can enable us to only share certain features?

**Comment by CloudeLecaw** [comment]
Maybe he can add something there in the CmiBungee Extansion like he made it with modules. :)
